### ISSUE TITLE
Pageant not used automatically

### ISSUE THREAD

**Issue by mensi** [ronf/asyncssh#215]
When using `asyncssh.connect('somehost')`, it looks like Pageant will never automatically be used. 

In `create_connection`, the following code is used to determine whether to load keys from an agent:

```
    if client_keys:
        client_keys = load_keypairs(client_keys, passphrase)
    elif client_keys == ():
        if agent_path:
            agent = yield from connect_agent(agent_path, loop=loop)

            if agent:
                client_keys = yield from agent.get_keys()
```
https://github.com/ronf/asyncssh/blob/master/asyncssh/connection.py#L5303

So an agent will only be connected to if an agent_path is set and client_keys is empty. However, in the case of Pageant on Windows, no agent_path is set since trying to talk to it is predicated on it being falsy:

```
        if not agent_path:
            try:
                _find_agent_window()
                transport = _PageantTransport()
```

https://github.com/ronf/asyncssh/blob/master/asyncssh/agent_win32.py#L167

The workaround is relatively simple, just use something like:

```
agent = await asyncssh.connect_agent()
client_keys = await agent.get_keys()
client = await asyncssh.connect('somehost', client_keys=client_keys)
```

As a library user, I would however expect this to work automatically. I assume the `if agent_path:` in `create_connection` could just be dropped so that the agent is unconditionally tried.

**Comment by ronf** [comment]
You're right - my intention here was to automatically try to use an SSH agent when one was available, as long as client_keys wasn't set. This works for non-Pageant agents when SSH_AUTH_SOCK is set (or agent_path is set explicitly), but the current code appears to be too aggressive about not attempting to connect to the agent at all when there's no agent path set, preventing Pageant from being used.

It looks like a quick fix for this would be to remove the "if agent_path" check in connection.py at line 5306 (in create_connection) that you identified above:

```patch
--- asyncssh/connection.py	2019-03-30 14:01:08.000000000 -0700
+++ asyncssh/connection.py.new	2019-05-28 19:44:48.000000000 -0700
@@ -5303,13 +5303,12 @@
     if client_keys:
         client_keys = load_keypairs(client_keys, passphrase)
     elif client_keys == ():
-        if agent_path:
-            agent = yield from connect_agent(agent_path, loop=loop)
+        agent = yield from connect_agent(agent_path, loop=loop)
 
-            if agent:
-                client_keys = yield from agent.get_keys()
-            else:
-                agent_path = None
+        if agent:
+            client_keys = yield from agent.get_keys()
+        else:
+            agent_path = None
 
         if not client_keys:
             client_keys = load_default_keypairs(passphrase)
```

While testing this, I think I may have also run across a problem with Pageant not supporting the alternate ssh-rsa signature algorithms. I had to add an explicit "signature_algs=['ssh-rsa']" argument to my connect call for it to work.

As an aside, I recently made a change to AsyncSSH to allow both agent keys and the default keys in the local filesystem to be loaded, rather than only loading the default local keys when the agent didn't return anything. That didn't change the behavior described here, but I thought I'd mention it as a behavior change you'll see in the next release.

Let me know if the above patch works if you have a chance to try it. In the meantime, I'll look into the signature_algs issue.

**Comment by ronf** [comment]
Actually, after a bit more testing here, I discovered that the above change is too aggressive. It'll try to use an agent even when you set agent_path=None explicitly. I'll post another patch to try shortly.

**Comment by ronf** [comment]
Ok - here's an alternate patch you can try:

```patch
--- asyncssh/connection.py	2019-03-30 14:01:08.000000000 -0700
+++ asyncssh/connection.py.new	2019-05-28 22:30:33.000000000 -0700
@@ -5298,12 +5298,12 @@
     agent = None
 
     if agent_path == ():
-        agent_path = os.environ.get('SSH_AUTH_SOCK', None)
+        agent_path = os.environ.get('SSH_AUTH_SOCK', ())
 
     if client_keys:
         client_keys = load_keypairs(client_keys, passphrase)
     elif client_keys == ():
-        if agent_path:
+        if agent_path is not None:
             agent = yield from connect_agent(agent_path, loop=loop)
 
             if agent:
```

In addition, I discovered that neither Pageant nor the Windows 10 OpenSSH agent seem to actually support the signature flags used to request RSA SHA2 signatures. So, you may need to also make the following change:

```patch
--- asyncssh/agent.py	2018-11-28 21:11:31.000000000 -0800
+++ asyncssh/agent.py.new	2019-05-28 22:35:12.000000000 -0700
@@ -124,7 +124,9 @@
         else:
             self.sig_algorithm = algorithm
 
-        if self.sig_algorithm == b'ssh-rsa':
+        # Neither Pageant nor the Win10 OpenSSH agent seems to support the
+        # ssh-agent protocol flags used to request RSA SHA2 signatures yet
+        if self.sig_algorithm == b'ssh-rsa' and sys.platform != 'win32':
             self.sig_algorithms = (b'rsa-sha2-256', b'rsa-sha2-512',
                                    b'ssh-rsa')
         else:
```

**Comment by ronf** [comment]
I just committed fixes c04e4d9a and 36c8992 to the "develop" branch with these changes (adapted to account for the new SSHClientConnectionOptions changes). This will be included in the next release. Thanks for the report!

**Comment by mensi** [comment]
I just gave the develop branch a try and can confirm it works as expected now :) Thanks for the quick fix!

**Comment by ronf** [comment]
No problem - thanks for the confirmation! I'll leave this open for now, closing it when I do the next release.

**Comment by ronf** [comment]
This fix is now available in release 1.17.0.
